src/App.js

```diff
@@ -1,24 +1,25 @@
-import logo from './logo.svg';
 import './App.css';
+import { Fragment } from 'react';
+import { Routes, Route } from 'react-router-dom';
+import TopNav from './components/TopNav';
+import Home from './components/Home';
+import CharClassBox from './components/CharClassBox';
+import UserSpellbook from './components/UserSpellbook';
+import SpellBox from './components/SpellBox';
+import SpellDetails from './components/SpellDetails';
 
-function App() {
+const App = () => {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <>
+      <TopNav />
+      <Routes>
+        <Route path="/" element={ <Home /> } />
+        <Route path="/class" element={ <CharClassBox /> } />
+        <Route path="/:class/spells" element={ <SpellBox /> } />
+        <Route path="/my-spellbook" element={ <UserSpellbook /> } />
+        <Route path="/spells/:index" element={ <SpellDetails /> }/>
+      </Routes>
```

Comment:
I'm really thankful that we sat down and worked this out together, because it really helped me clear up some of the fuzzy bits about router.  It also helps me to make sense of some of the API path naming conventions.  10/10 ⭐️

Reply:
Agreed!